kedro-datasets/pyproject.toml

```
@@ -21,7 +21,7 @@ api-apidataset = ["requests~=2.20"]
 biosequence = ["kedro-datasets[biosequence.BioSequenceDataSet]"]
 biosequence-biosequencedataset = ["biopython~=1.73"]
 dask = ["kedro-datasets[dask.ParquetDataSet]"]
-dask-parquetdataset = ["dask[complete]~=2021.10", "triad>=0.6.7, <1.0"]
+dask-parquetdataset = ["dask[complete]~=2023.7", "triad>=0.6.7, <1.0"]
```

What about a more flexible pinning, like

```suggestion
dask-parquetdataset = ["dask[complete]>=2021.10", "triad>=0.6.7, <1.0"]
```

? Dask doesn't follow SemVer https://github.com/dask/community/issues/100 so this pinning is not very useful - breaking changes might be introduced at any time. We can shift the onus to the user on maintaining an appropriate pinning if something breaks. Which I don't expect to, because we use pretty fundamental Dask APIs anyway.

---

Thank you for your suggestion. I agree that a more flexible pinning approach would be beneficial given Dask's versioning policy. Shifting the responsibility to the user for maintaining an appropriate pinning seems reasonable.